tests/tests_integration/test_logging.py

```diff
@@ -0,0 +1,404 @@
+
+import pytest
+import test_utils
+
+from datashuttle.datashuttle import DataShuttle
+from datashuttle.utils import ds_logger
+
+BAD_FILECHAR = "?" if platform.system() == "Windows" else "/"
+
+
+class TestCommandLineInterface:
+    @pytest.fixture(scope="function")
+    def clean_project_name(self):
+        """
+        Create an empty project, but ensure no
+        configs already exists, and delete created configs
+        after test.
+        """
+        project_name = "test_logging"
+        test_utils.delete_project_if_it_exists(project_name)
+        yield project_name
+        test_utils.delete_project_if_it_exists(project_name)
+
+    @pytest.fixture(scope="function")
+    def setup_project(self, tmp_path):
+        """
+        Setup a project with default configs to use
+        for testing.
+
+        # Note this fixture is a duplicate of project()
+        in test_filesystem_transfer.py fixture
+        """
+        test_project_name = "test_logging"
+        setup_project, cwd = test_utils.setup_project_fixture(
+            tmp_path, test_project_name
+        )
+
+        self.delete_log_files(setup_project._logging_path)
+
+        yield setup_project
+        test_utils.teardown_project(cwd, setup_project)
+
+    # ----------------------------------------------------------------------------------------------------------
+    # Test Public API Logging
+    # ----------------------------------------------------------------------------------------------------------
+
+    def read_log_file(self, logging_path):
+
+        log_filepath = glob.glob(str(logging_path / "*.log"))
+
+        assert len(log_filepath) == 1, (
+            f"there should only be one log "
+            f"in log output path {logging_path}"
+        )
+        log_filepath = log_filepath[0]
+
+        with open(log_filepath, "r") as file:
+            log = file.read()
+
+        return log
+
+    def delete_log_files(self, logging_path):
+        ds_logger.close_log_filehandler()
+        logs = glob.glob((str(logging_path / "*.log")))
+        for log in logs:
+            os.remove(log)
+
+    def test_logs_make_config_file(self, clean_project_name, tmp_path):
+        """"""
+        project = DataShuttle(clean_project_name)
+
+        project.make_config_file(
+            tmp_path, "two", "local_filesystem", use_behav=True
+        )
+
+        log = self.read_log_file(project._logging_path)
+
+        assert "Starting make_config_file" in log
+        assert "Successfully created rclone config." in log
+        assert (
+            "Configuration file has been saved and options loaded into datashuttle."
+            in log
+        )
+        assert "Update successful. New config file:" in log
+
+    def test_logs_update_config(self, setup_project):
+
+        setup_project.update_config("remote_path", "test_path")
+
+        log = self.read_log_file(setup_project._logging_path)
+
+        assert "Starting update_config" in log
```

Comment:
Is it worth having a "validation" log file in with the tests? May be easier to load & compare rather than checking line by line here.

Reply:
This is nice idea, unfortunately there seems to be some cross-platform differences in the log output and so selected randomly some lines that are the same cross platform. It's true though that these are completely arbitrary lines. The bar here was 'is printing a log that is describing the correct process' but these tests could be stricter.